backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/invoice/C_Invoice_StepDef.java

```diff
@@ -88,20 +113,32 @@ public void validate_created_invoices(@NonNull final DataTable table)
 		final List<Map<String, String>> dataTable = table.asMaps();
 		for (final Map<String, String> row : dataTable)
 		{
-			final String identifier = DataTableUtil.extractStringForColumnName(row, I_C_Invoice.COLUMNNAME_C_Invoice_ID + "." + StepDefConstants.TABLECOLUMN_IDENTIFIER);
+			final String identifier = DataTableUtil.extractStringForColumnName(row, I_C_Invoice.COLUMNNAME_C_Invoice_ID + "." + TABLECOLUMN_IDENTIFIER);
 
 			final I_C_Invoice invoice = invoiceTable.get(identifier);
 
 			validateInvoice(invoice, row);
 		}
 	}
 
+	@Then("^enqueue invoice candidate for invoicing and after not more than (.*)s, the invoice is found$")
+	public void generateInvoiceFromInvoiceCand(final int timeoutSec, @NonNull final DataTable dataTable) throws InterruptedException
+	{
+		for (final Map<String, String> row : dataTable.asMaps())
+		{
+			final String invoiceCandIdentifier = DataTableUtil.extractStringForColumnName(row, COLUMNNAME_C_Invoice_Candidate_ID + "." + TABLECOLUMN_IDENTIFIER);
+			final I_C_Invoice_Candidate invoiceCandidate = invoiceCandidateTable.get(invoiceCandIdentifier);
+
+			enqueueInvoiceCand(invoiceCandidate, row, timeoutSec);
+		}
+	}
+
 	@Then("^enqueue candidate for invoicing and after not more than (.*)s, the invoice is found$")
 	public void generateInvoice(final int timeoutSec, @NonNull final DataTable table) throws InterruptedException
 	{
 		final Map<String, String> row = table.asMaps().get(0);
```

Comment:
I think, since this is a table with just one col and row, better change the step-dev to sth like "^enqueue invoice candidate of order (.*) for invoicing and after not more than (.*)s, the invoice is found$"

Reply:
refactored accordingly